src/topology/algebra/group.lean

```diff
@@ -241,6 +232,114 @@ lemma continuous_within_at.inv (hf : continuous_within_at f s x) :
+  refine is_closed_Inter (λ i, is_closed_eq (continuous_apply _) (continuous_apply _).inv),
+end
+
+end pointwise_limits
+
+instance additive.has_continuous_neg [h : topological_space H] [has_inv H]
+  [has_continuous_inv H] : @has_continuous_neg (additive H) h _ :=
+{ continuous_neg := @continuous_inv H _ _ _ }
+
+instance multiplicative.has_continuous_inv [h : topological_space H] [has_neg H]
+  [has_continuous_neg H] : @has_continuous_inv (multiplicative H) h _ :=
+{ continuous_inv := @continuous_neg H _ _ _ }
+
+end continuous_inv
+
+section lattice_ops
+
+variables {ι' : Sort*} [has_inv G] [has_inv H] {ts : set (topological_space G)}
+  (h : Π t ∈ ts, @has_continuous_inv G t _) {ts' : ι' → topological_space G}
+  (h' : Π i, @has_continuous_inv G (ts' i) _) {t₁ t₂ : topological_space G}
+  (h₁ : @has_continuous_inv G t₁ _) (h₂ : @has_continuous_inv G t₂ _)
+  {t : topological_space H} [has_continuous_inv H]
+
+
+@[to_additive] lemma has_continuous_inv_Inf :
+  @has_continuous_inv G (Inf ts) _ :=
+{ continuous_inv := continuous_Inf_rng (λ t ht, continuous_Inf_dom ht
+  (@has_continuous_inv.continuous_inv G t _ (h t ht))) }
+
+include h'
+
+@[to_additive] lemma has_continuous_inv_infi :
+  @has_continuous_inv G (⨅ i, ts' i) _ :=
+by {rw ← Inf_range, exact has_continuous_inv_Inf (set.forall_range_iff.mpr h')}
+
+omit h'
+
+include h₁ h₂
+
+@[to_additive] lemma has_continuous_inv_inf :
+  @has_continuous_inv G (t₁ ⊓ t₂) _ :=
+by {rw inf_eq_infi, refine has_continuous_inv_infi (λ b, _), cases b; assumption}
+
+end lattice_ops
+
+section topological_group
+
+/-!
+### Topological groups
+
+A topological group is a group in which the multiplication and inversion operations are
+continuous. Topological additive groups are defined in the same way. Equivalently, we can require
+that the division operation `λ x y, x * y⁻¹` (resp., subtraction) is continuous.
+-/
+
+/-- A topological (additive) group is a group in which the addition and negation operations are
+continuous. -/
+class topological_add_group (G : Type u) [topological_space G] [add_group G]
+  extends has_continuous_add G, has_continuous_neg G : Prop
+
+/-- A topological group is a group in which the multiplication and inversion operations are
+continuous. -/
+@[to_additive]
+class topological_group (G : Type*) [topological_space G] [group G]
+  extends has_continuous_mul G, has_continuous_inv G : Prop
+
+variables [topological_space G] [group G] [topological_group G]
+[topological_space α] {f : α → G} {s : set α} {x : α}
+
+export has_continuous_inv (continuous_inv)
+export has_continuous_neg (continuous_neg)
+
+/-- Conjugation in a topological group is continuous.-/
+@[to_additive "Conjugation in a topological additive group is continuous."]
+lemma topological_group.continuous_conj (g : G) : continuous (λ (h : G), g * h * g⁻¹) :=
+(continuous_mul_right g⁻¹).comp (continuous_mul_left g)
+
 @[to_additive]
 lemma is_compact.inv {s : set G} (hs : is_compact s) : is_compact (s⁻¹) :=
```

Comment:
Does this need `topological_group` instead of just `has_continuous_inv`?

Reply:
almost? you still need both `has_continuous_mul` and `has_continuous_inv`, but I guess you don't technically need it to be a group.

Reply:
Where is `has_continuous_mul` needed for `is_compact.inv`?

Reply:
Sorry, I thought it was the highlighted bit about `topological_group.continuous_conj`.

Reply:
I guess it makes sense to generalize `continuous_conj` too though, assuming it's true on `ennreal`?

Reply:
Okay, I generalized this, but then realized the stated version was conjugation by a fixed element, and therefore didn't need `has_continuous_inv`. I have added some alternative versions too. Please have a look and let me know about naming or anything else. I kind of wanted to keep them in the `topological_group` namespace for two reasons (a) this is where I would expect people to look for them and (b) then we don't have to go hunting for renaming (for `topological_group.continuous_conj` that is). By the way, I realize the first one can prove the latter two, but for `continuous_conj` we don't need the type class, and for `continuous_conj'` it's just easier to prove directly.